pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -102,22 +115,30 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 	sst.Count = toputils.BytesToUint64LittleEndian(fdata[idx : idx+8])
 	idx += 8
 
-	hllSize := toputils.BytesToUint16LittleEndian(fdata[idx : idx+2])
-	idx += 2
+	var hllSize uint32
 
 	switch version {
-	case utils.VERSION_SEGSTATS[0]:
+	case utils.VERSION_SEGSTATS_BUF[0]:
+		hllSize = toputils.BytesToUint32LittleEndian(fdata[idx : idx+4])
+		idx += 4
+	case utils.VERSION_SEGSTATS_BUF_LEGACY_2[0], utils.VERSION_SEGSTATS_BUF_LEGACY_1[0]:
+		hllSize = uint32(toputils.BytesToUint16LittleEndian(fdata[idx : idx+2]))
+		idx += 2
+	default:
```

Comment:
Why don't you handle `VERSION_SEGSTATS_BUF_LEGACY_1`? Even if you handle it just by throwing an error because that version is no longer supported, I think that will be more useful/clear to someone debugging than it will be if you let it fall to the `default` case

Reply:
I am handling that error case below here: https://github.com/siglens/siglens/pull/1564/files#diff-93111a0cc266401b8aa0d7c5db0fa5cf6ef706e35b8b4ed2d54c0e54459a9c98R132-R134

This is just after the switch case and in the switch case I am reading the `hllSize` for both legacy 1 and legacy 2 versions in the same case. As they are both have size to `2 bytes`. 
And we don't want to error out even for `legacy 1` cause we want to read other segstats.
The corresponding code:

```
if version == utils.VERSION_SEGSTATS_BUF_LEGACY_1[0] {
		log.Infof("qid=%d, readSingleSst: ignoring Hll (old version)", qid)
	}
```

Reply:
Oh, my bad. This looks good